python/pyspark/pandas/tests/data_type_ops/test_boolean_ops.py

```diff
@@ -226,10 +226,10 @@ def test_rpow(self):
 
         b_pser, b_psser = pdf["bool"], psdf["bool"]
         # float is returned always in pandas-on-Spark
-        self.assert_eq((1 ** b_pser).astype(float), 1 ** b_psser)
-        self.assert_eq(0.1 ** b_pser, 0.1 ** b_psser)
+        self.assert_eq((1**b_pser).astype(float), 1**b_psser)
+        self.assert_eq(0.1**b_pser, 0.1**b_psser)
         self.assertRaises(TypeError, lambda: "x" ** b_psser)
-        self.assertRaises(TypeError, lambda: True ** b_psser)
+        self.assertRaises(TypeError, lambda: True**b_psser)
         self.assertRaises(TypeError, lambda: datetime.date(1994, 1, 1) ** b_psser)
         self.assertRaises(TypeError, lambda: datetime.datetime(1994, 1, 1) ** b_psser)
```

Comment:
I'm not sure why Black doesn't reformat this, too...

Shouldn't it be reformatted like:

```diff
-         self.assertRaises(TypeError, lambda: datetime.date(1994, 1, 1) ** b_psser)
-         self.assertRaises(TypeError, lambda: datetime.datetime(1994, 1, 1) ** b_psser)
+         self.assertRaises(TypeError, lambda: datetime.date(1994, 1, 1)**b_psser)
+         self.assertRaises(TypeError, lambda: datetime.datetime(1994, 1, 1)**b_psser)
```

???

Reply:
I'm no black expert, but from the discussion I had with @HyukjinKwon above, it looks like it's a operator precedence behavior thing.

> If operators with different priorities are used, consider adding whitespace around the operators with the lowest priority(ies). Use your own judgment; however, never use more than one space, and always have the same amount of whitespace on both sides of a binary operator:

in 

```
self.assertRaises(TypeError, lambda: datetime.date(1994, 1, 1) ** b_psser)
```

The binary operator `**` is not paired with a lower priority other operator.

Reply:
Yeah, I understand about the priority but just wonder why Black did:

```diff
-         self.assertRaises(TypeError, lambda: True ** b_psser)
+         self.assertRaises(TypeError, lambda: True**b_psser)
```

but didn't

```diff
-         self.assertRaises(TypeError, lambda: datetime.date(1994, 1, 1) ** b_psser)
-         self.assertRaises(TypeError, lambda: datetime.datetime(1994, 1, 1) ** b_psser)
+         self.assertRaises(TypeError, lambda: datetime.date(1994, 1, 1)**b_psser)
+         self.assertRaises(TypeError, lambda: datetime.datetime(1994, 1, 1)**b_psser)
```

It seems like just inconsistence behavior from Black, so maybe we can just keep is as is for now.

Reply:
The reason is explained here https://github.com/apache/spark/pull/37305#discussion_r931339075 - in short the operators are not simple ( `True` vs `datetime.datetime(1994, 1, 1)`)